app/views/overrides/activation_keys/_host_environment_select.html.erb

```diff
@@ -34,10 +34,8 @@ end %>
   <input type="hidden" name="host[content_facet_attributes][id]" value="<%= @host.content_facet.try(:id) %>">
 <% end %>
 
-<% proxies = SmartProxy.with_content.unscoped.with_taxonomy_scope(@location,@organization,:path_ids) %>
-<% if proxies.count > 0 %>
-    <%= select_f f, :content_source_id, proxies, :id, :name,
-             { :include_blank => blank_or_inherit_f(f, :content_source) },
-             { :label       => _("Content Source"),
-               :help_inline => _("Use this as a source for installation and updates.") } %>
-<% end %>
+<% proxies = accessible_resource_records(:smart_proxy).with_content.with_taxonomy_scope(@location,@organization,:path_ids) %>
```

Comment:
Note that you also need to include the current proxy even if the user is not allowed to see it to prevent limited users from changing the setting unknowingly, see https://github.com/theforeman/foreman/pull/3369/files#diff-49521b3e6ed59e866e058b10bca549fdR44 for how this is handled for puppet proxy.  This might also be relevant for the other fields, not sure on the logic behind `.readable`.
Also, `accessible_resource_records` already adds the taxonomy scope so no need to do it here.